ee/mpeg/src/libmpeg_core.c

```diff
@@ -0,0 +1,735 @@
+        if (ret == 0x23) {
+            mbai += 0x21;
+        }
+
+        if (ret < 0x22) {
+            mbai += ret;
+            break;
+        }
+    }
+
+    s_BitsBuffered = 32;
+    s_LocalBits    = *R_EE_IPU_TOP;
+    return mbai;
+}
+
+int _MPEG_GetMBType(void)
+{
+    _ipu_sync();
+    *R_EE_IPU_CMD = IPU_COMMAND_VDEC | IPU_COMMAND_VDEC_MBTYPE;
+    u32 ret       = _ipu_sync_data();
+    if (ret != 0) {
+        ret &= 0xffff;
+        s_BitsBuffered = 32;
+        s_LocalBits    = *R_EE_IPU_TOP;
+    }
+    return ret;
+}
+
+int _MPEG_GetMotionCode(void)
+{
+    short mcode;
+    u32 ret;
+
+    _ipu_sync();
+    *R_EE_IPU_CMD = IPU_COMMAND_VDEC | IPU_COMMAND_VDEC_MCODE;
+    ret           = _ipu_sync_data();
+
+    if (ret == 0) {
+        return -32768;
+    }
+
+    s_BitsBuffered = 32;
+    s_LocalBits    = *R_EE_IPU_TOP;
+
+    mcode = (short)ret;
+
+    return mcode;
+}
+int _MPEG_GetDMVector(void)
+{
+    u32 ret;
+
+    _ipu_sync();
+    *R_EE_IPU_CMD  = IPU_COMMAND_VDEC | IPU_COMMAND_VDEC_DMV;
+    ret            = _ipu_sync_data() & 0xffff;
+    s_BitsBuffered = 32;
+    s_LocalBits    = *R_EE_IPU_TOP;
+    return ret;
+}
+
+void _MPEG_SetIDCP(void)
+{
+    unsigned int var1 = _MPEG_GetBits(2);
+    *R_EE_IPU_CTRL    = (*R_EE_IPU_CTRL & ~0x30000) | var1 << 0x10;
+}
+
+void _MPEG_SetQSTIVFAS(void)
+{
+    unsigned int qst = _MPEG_GetBits(1);
+    unsigned int ivf = _MPEG_GetBits(1);
+    unsigned int as  = _MPEG_GetBits(1);
+    *R_EE_IPU_CTRL   = (*R_EE_IPU_CTRL & ~0x700000) | qst << 22 | ivf << 21 | as << 20;
+}
+
+void _MPEG_SetPCT(unsigned int arg0)
+{
+    _ipu_sync();
+    *R_EE_IPU_CTRL = (*R_EE_IPU_CTRL & ~0x07000000) | arg0 << 0x18;
+}
+
+void _MPEG_BDEC(int mbi, int dcr, int dt, int qsc, void *spaddr)
+{
+    *R_EE_D3_MADR = ((u32)spaddr & ~0xf0000000) | 0x80000000;
+    *R_EE_D3_QWC  = 0x30;
+    *R_EE_D3_CHCR = 0x100;
+    _ipu_sync();
+    *R_EE_IPU_CMD = IPU_COMMAND_BDEC | mbi << 27 | dcr << 26 | dt << 25 | qsc << 16;
+}
+
+int _MPEG_WaitBDEC(void)
+{
+    int oldintr;
+
+    while (1) {
+        _ipu_sync();
+        if ((*s_pEOF != 0)) {
+            break;
+        }
+
+        if (*R_EE_IPU_CTRL & IPU_CTRL_ECD) {
+            break;
+        }
+
+        if (*R_EE_D3_QWC == 0) {
+            s_BitsBuffered = 32;
+            s_LocalBits    = *R_EE_IPU_TOP;
+            return 1;
+        }
+    }
+
+    _ipu_suspend();
+    *R_EE_IPU_CTRL = IPU_CTRL_RST;
+    _ipu_resume();
+
+    oldintr         = DIntr();
+    *R_EE_D_ENABLEW = *R_EE_D_ENABLER | 0x10000;
+    *R_EE_D3_CHCR   = 0;
+    *R_EE_D_ENABLEW = *R_EE_D_ENABLER & ~0x10000;
+    if (oldintr) {
+        EIntr();
+    }
+    *R_EE_D3_QWC   = 0;
+    s_BitsBuffered = 0;
+    s_LocalBits    = 0;
+    return 0;
+}
+
+void _MPEG_dma_ref_image(_MPEGMacroBlock8 *mb, _MPEGMotion *motions, s64 n_motions, int width)
+{
+    const u32 mbwidth = width * sizeof(*mb);
+    qword_t *q;
+    int i;
+
+    if (n_motions > 0) {
+        while (*R_EE_D9_CHCR & 0x100)
```

Comment:
Move q and i declarations inside this block.

(Alternatively, invert the if condition and make it return, then move the contents of the block out of the if statement)